tuf/repository_lib.py

```diff
@@ -1541,6 +1559,168 @@ def _get_hashes_and_length_if_needed(use_length, use_hashes, full_file_path,
 
 
 
+
+# I couldn't find a currently maintained python library for this, so
+# implementing it here. It would be better to implement this in c,
+# even better to use an existing library
+# This is inspired by: https://www.literateprograms.org/miller-rabin_primality_test__python_.html
+def miller_rabin_round(a, s, d, n):
+  a_to_power = pow(a, d, n)
+  if a_to_power == 1:
+    return True
+  for i in range(s):
+    if a_to_power == n - 1:
+      return True
+    a_to_power = (a_to_power * a_to_power) % n
+  return False
+
+
+
+
+
+def miller_rabin(n, rounds):
+  if n == 1:
+    return false
+  if n == 2 or n == 3:
+    return true
+
+  d = n -1
+  s = 0
+  while d % 2 == 0:
+    d = d >> 1
+    s = s + 1
+
+  for i in range(rounds):
+    a = random.randrange(n)
+    if not miller_rabin_round(a, s, d, n):
+      return False
+  return True
+
+
+
+
+# RSA Accumulator code insprired by https://github.com/ElrondNetwork/elrond-go/blob/v1.0.30/crypto/accumulator/rsa/rsaAcc.go
+def hash_to_prime(data):
+  # TODO: move constant definitions
+  basesMillerRabin = 12
+
+  h = blake2b(str(data).encode('utf-8'))
+  p = int(h.hexdigest(), 16)
+
+  # use Miller-Rabin primality test, if p is not prime, do more rounds of hashing
+  while (not miller_rabin(p, basesMillerRabin)):
+    h = blake2b(str(p).encode('utf-8'))
+    p = int(h.hexdigest(), 16)
+
+  return p
+
+
+
+class acc_contents(object):
+  contents = None
+  name = None
+  proof = None
+
+  def __init__(self, name, contents):
+    # Include the name to ensure the digest differs between elements and cannot be replayed
+    contents["name"] = name
+    self.contents = contents
+    self.name = name
+
+  def set_proof(self, proof):
+    self.proof = proof
+
+
+
+def _build_rsa_acc(fileinfodict):
+  """
+  Create an RSA accululator from the snapshot fileinfo and writes it to individual snapshot files
+
+  Returns the root and leaves
+  """
+
+  # RSA accululator contants
+  # TODO: move constant definitions
+  g = 3
+  # Modulus from https://en.wikipedia.org/wiki/RSA_numbers#RSA-2048
+  # We will want to generate a new one
```

Comment:
This might make sense as a setting and/or part of the format---not everyone should use ours.

(I think this is what you're saying, just trying to make explicit)